firmware/hw_layer/stepper_dual_hbridge.cpp

```diff
@@ -22,12 +22,26 @@ static const int8_t phaseB[] =
     -1
 };
 
+static const int8_t microPhases[] = {
+    0,    20,  38,  56,  71,  83,  92,  98,
+    100,  98,  92,  83,  71,  56,  38,  20,
+    0,   -20, -38, -56, -71, -83, -92, -98,
+   -100, -98, -92, -83, -71, -56, -38, -20
+};
+
+static const int maxNumSteps = 8;
+static const int tableSizeMask = (sizeof(microPhases) / sizeof(int8_t) - 1);
+static const float phaseDutyCycleDivisor = 1 / (100.0f * 100.0f);	// both the phase degrees and duty cycle settings are in %
+
+
 void DualHBridgeStepper::initialize(DcMotor* motorPhaseA, DcMotor* motorPhaseB, float reactionTime)
 {
     setReactionTime(reactionTime);
 
     m_motorPhaseA = motorPhaseA;
     m_motorPhaseB = motorPhaseB;
+
+    efiAssertVoid(CUSTOM_ERR_ASSERT, CONFIG(stepperNumMicroSteps) <= maxNumSteps, "stepperNumMicroSteps");
 }
 
 bool DualHBridgeStepper::step(bool positive) {
```

Comment:
This isn't actually micro stepping.... it's doing N micro steps per full step, but never less than one full step worth of movement.

Right now we still do full integer steps, not micro steps.  Should we instead represent as N * M steps, where N is the number of micro steps, and M is the number of actual steps from stop-to-stop on the stepper? Then setting positions 100 and 101 would be one full step apart, but in microstep mode, positions 100.25, 100.5, 100.75 would also exist between those positions.  The idle controller already emits a floating point position, so this wouldn't be too hard.

Reply:
Yes, exactly. Although this PR has already some noticeable advantages (less noise, motion smoothness, current chopping), it's also a preparation to utilize the fractional steps in the future. It should be added later in a separate PR.